packages/dds/sequence/src/test/intervalIndexTestUtils.ts

```diff
@@ -68,3 +70,141 @@ export function generateRandomIntervals(options: RandomIntervalOptions) {
 
 	return intervals;
 }
+
+/**
+ * Infer the side of a SequencePlace.
+ */
+export function sideFromSequencePlace(place: SequencePlace): Side {
+	if (typeof place === "object") {
+		return place.side;
+	} else if (place === "start") {
+		return Side.After;
+	} else {
+		return Side.Before;
+	}
+}
+/**
+ * position identifies a character, and does not take into account which side of the character.
+ *
+ * @param place - The SequencePlace used as an expected position of a sequence endpoint.
+ *
+ * @param ifEnd - The position to expect if `place` is "end". This is not currently consistent.
+ * Sometimes it is string.length, sometimes it is -1.
+ */
+export function expectedPositionFromSequencePlace(
+	place: SequencePlace,
+	ifEnd: number = -1,
+): number {
+	if (place === "start") {
+		return 0;
+	} else if (place === "end") {
+		return ifEnd;
+	} else if (typeof place === "object") {
+		return place.pos;
+	} else {
+		return place;
+	}
+}
+
+/**
+ * @returns the index to be used with methods such as substring, taking side into account.
+ */
+export function expectedIndexFromSequencePlace(place: SequencePlace, ifEnd: number = -1): number {
+	if (place === "start") {
+		return 0;
+	} else if (place === "end") {
+		return ifEnd;
+	} else if (typeof place === "object") {
+		return place.side === Side.Before ? place.pos : place.pos + 1;
+	} else {
+		return place;
+	}
+}
+
+/**
+ * Resolves an interval in a given shared string to indices which may be passed to methods such as substring.
+ * In other words, it re-anchors the endpoints to both use Side.before.
+ *
+ * @param sharedString - The shared string to resolve `interval` in.
+ * This must be the shared string which created the interval.
+ *
+ * @param interval - The interval to resolve.
+ *
+ * @returns the indices to be used with methods such as substring for a given SequenceInterval.
+ */
+export function indicesFromSequenceInterval(
+	sharedString: ISharedString,
+	interval: SequenceInterval,
+): [number, number] {
+	let start = sharedString.localReferencePositionToPosition(interval.start);
+	if (start === -1) {
+		start = sharedString.getLength();
+	}
+	const end = sharedString.localReferencePositionToPosition(interval.end);
+	return [
+		interval.startSide === Side.Before ? start : start + 1,
+		interval.endSide === Side.Before ? end : end + 1,
+	];
+}
+
+/**
+ * Assert that the interval with the given id in the given shared string matches the expected interval.
+ * @param sharedString - The shared string to check the interval in.
+ * @param intervalId - The id of the interval to check.
+ * @param expected - The expected interval.
+ */
+export function assertInterval(
```

Comment:
If I'm remembering right, I think we have an existing assertIntervals function that doesn't have the side comparisons currently. It might be nice to try to unify the comparisons, or reuse anything if possible.

Reply:
Existing usages of `assertSequenceIntervals` don't specify the stickiness or endpoint sides they expect, so unifying would require updating existing usages. Unifying `assertInterval` and `assertSequenceInterval` should be done in a separate PR.